src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/ObjectWithParameterizedConstructorConverter.cs

```diff
@@ -184,7 +184,7 @@ internal sealed override bool OnTryRead(ref Utf8JsonReader reader, Type typeToCo
 
                         if (dataExtKey == null)
                         {
-                            jsonPropertyInfo.SetExtensionDictionaryAsObject(obj, propValue);
+                            jsonPropertyInfo.SetValueAsObject(obj, propValue, ref state);
```

Comment:
Continuing on the comment from https://github.com/dotnet/runtime/pull/72937/files#r931340334, I think we could completely eliminate the two methods with somthing like the following:
```suggestion
                            Debug.Assert(jsonPropertyInfo.Set != null);
                            jsonPropertyInfo.Set(obj, propValue);
                            state.Current.MarkRequiredPropertyAsRead(jsonPropertyInfo);
```